contracts/contracts/GrantRoundManager.sol

```diff
@@ -204,15 +198,15 @@ contract GrantRoundManager {
       }
 
       // Otherwise, execute swap
-      ISwapRouter.ExactInputParams memory params = ISwapRouter.ExactInputParams(
+      ExactInputParams memory params = ExactInputParams(
         _swaps[i].path,
         address(this), // send output to the contract and it will be transferred later
         _deadline,
         _swaps[i].amountIn,
         _swaps[i].amountOutMin
       );
       uint256 _value = _tokenIn == WETH && msg.value > 0 ? msg.value : 0;
-      swapOutputs[_tokenIn] = router.exactInput{value: _value}(params); // save off output amount for later
+      swapOutputs[_tokenIn] = this.exactInput{value: _value}(params); // save off output amount for later
```

Comment:
`this.exactInput` makes an external call back to this contract, which mainly just wastes gas. Instead let's change the `exactInput` method from `external` to `public` so we can remove the `this.` keyword. Doing that means we'll also be able to remove the `value` passed with the call, since that's only needed since we're forwarding the ETH to an external call

Reply:
tried this, but errors out on function call options (used for external functions, not public functions)

Reply:
would it make sense to stick to using `this` and leave a TODO to tweak the exactInput into public function and fix the other things in the custom SwapContract which break as a result ?

Reply:
Hmm, my initial reaction is that it shouldn't be much more work to do things the "right" way here, but if keeping `this` is a lot simpler then I'm ok with that for now

Reply:
yes we're all good here - thanks for the input though thelostone-mc